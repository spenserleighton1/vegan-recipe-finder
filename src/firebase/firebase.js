import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore'

const config = {
    apiKey: "AIzaSyDfChryFf9ERaEQz2RgyyN4VzkPKAEzuqY",
    authDomain: "vegan-recipe-finder.firebaseapp.com",
    databaseURL: "https://vegan-recipe-finder.firebaseio.com",
    projectId: "vegan-recipe-finder",
    storageBucket: "",
    messagingSenderId: "1056656932624"
  };

firebase.initializeApp(config);

export const auth = firebase.auth();
export const firestore = firebase.firestore();
export const docRef = firestore.doc("users/savedRecipes");

export default { auth, docRef };