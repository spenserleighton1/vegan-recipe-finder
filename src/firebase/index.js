import * as auth from './auth';
import * as firebase from './firebase';

import { docRef } from './firebase'


export {
  auth,
  firebase,
  docRef
};