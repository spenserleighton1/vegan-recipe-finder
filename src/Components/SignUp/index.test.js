import React from 'react';
import { shallow } from 'enzyme';
import { SignUpForm } from './';
import { auth } from '../firebase'

describe('SignUpForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignUpForm />)
  })

  it('should start with default state of username,. email, pWord1 pWord2, and error', () => {
    const actual = wrapper.state()
    const expected = {email: "", error: null, passwordOne: "", passwordTwo: "", username: ""}

    expect(actual).toEqual(expected)
  })

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: {name: 'email', value: 'sleepy' }}

    wrapper.instance().handleChange(mockEvent)

    const actual = wrapper.state('email')

    expect(actual).toEqual('sleepy')
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
