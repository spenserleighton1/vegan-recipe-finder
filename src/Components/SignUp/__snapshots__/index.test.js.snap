// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SignUp
</h1>,
        <form
          className="signup-form"
          onSubmit={[Function]}
>
          Username:
          <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Full Name"
                    type="text"
                    value=""
          />
          Email:
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          Password:
          <input
                    name="passwordOne"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          Confirm Password:
          <input
                    name="passwordTwo"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
          <p />
</form>,
      ],
      "className": "sign-up",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SignUp",
        },
        "ref": null,
        "rendered": "SignUp",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Username:",
            <input
              name="username"
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
              value=""
/>,
            "Email:",
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            "Password:",
            <input
              name="passwordOne"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            "Confirm Password:",
            <input
              name="passwordTwo"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
            <p />,
          ],
          "className": "signup-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Username:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "Full Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Email:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Password:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordOne",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Confirm Password:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordTwo",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SignUp
</h1>,
          <form
            className="signup-form"
            onSubmit={[Function]}
>
            Username:
            <input
                        name="username"
                        onChange={[Function]}
                        placeholder="Full Name"
                        type="text"
                        value=""
            />
            Email:
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
                        value=""
            />
            Password:
            <input
                        name="passwordOne"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            Confirm Password:
            <input
                        name="passwordTwo"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        type="password"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign Up
            </button>
            <p />
</form>,
        ],
        "className": "sign-up",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SignUp",
          },
          "ref": null,
          "rendered": "SignUp",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Username:",
              <input
                name="username"
                onChange={[Function]}
                placeholder="Full Name"
                type="text"
                value=""
/>,
              "Email:",
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
/>,
              "Password:",
              <input
                name="passwordOne"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              "Confirm Password:",
              <input
                name="passwordTwo"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign Up
</button>,
              <p />,
            ],
            "className": "signup-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Username:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "username",
                "onChange": [Function],
                "placeholder": "Full Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordOne",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Confirm Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordTwo",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
